src/stores/api.js

```diff
@@ -23,4 +27,39 @@ export async function getAllTimeGreat(lastThirtyDays=false) {
     raw: true,
   });
   return res.data;
+}
+
+export async function getLoggedInUser() {
+  try {
+    const res = await axios.get(BASE_URL + "/api/user", { raw: true });
+    const username = res.data["username"];
+    localStorage.setItem("currentUser", username);
+  } catch (error) {
+    console.log(error);
+  }
+}
+
+export async function isUserLoggedIn() {
+  const currentUser = localStorage.getItem("currentUser");
+  if (currentUser) {
+    return currentUser;
+  }
+  return;
+}
+
+export async function LogIn() {
+  if (!localStorage.getItem("currentUser")) {
+    await getLoggedInUser();
+  }
+
+  if (!localStorage.getItem("currentUser")) {
+    window.open("http://localhost:8082/api/login");
```

Comment:
Why use `window.open` here and not something like `window.location.href` to redirect to `/api/login`?

I'm not a huge fan of `window.open`, because you end up with two browser windows open to Toolhunt.

Reply:
I will look into the difference between these two and update.

> I'm playing around with this some more this morning. I'm feeling like using localStorage is a bad idea; our Toolhub session expires pretty much anytime anything happens (e.g., a page refresh) but the localStorage data persists, so in terms of component rendering, the page acts as though a user is logged in when really they're not.
> 
> (Another problem with localStorage: anyone with the ability to open the console can edit what's in there, so we could never trust the information we're getting from it.)


So the reason for localStorage is to be able to make isUserLoggedIn checks
for example, before a user is allowed to submit a task/ or see dashboard metrics, they should be logged in - I would prefer to not call the backend to make this check, I should be able to check some data on the front-end.

what do you think?